src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -788,30 +797,108 @@ private String getRouterId() {
+            manifest.uploadGraph = true;
+            manifest.uploadGraphBuildLogs = true;
+            manifest.uploadGraphBuildReport = true;
+            if (deployment.buildGraphOnly || otpServer.ec2Info.hasSeparateGraphBuildConfig()) {
+                // This instance should be ran to only build the graph
+                manifest.runServer = false;
+            } else {
+                // This instance will both run a graph and start the OTP server
+                manifest.routerConfigJSON = deployment.generateRouterConfigAsString();
+                manifest.runServer = true;
+                manifest.uploadServerStartupLogs = true;
+            }
+        } else {
+            // This instance will only start the OTP server with a prebuilt graph
+            manifest.buildGraph = false;
+            manifest.routerConfigJSON = deployment.generateRouterConfigAsString();
+            manifest.runServer = true;
+            manifest.uploadServerStartupLogs = true;
+        }
+        String otpRunnerManifestFile = String.format("/var/%s/otp-runner-manifest.json", getTripPlannerString());
+
+        //////////////// BEGIN USER DATA
+        List<String> lines = new ArrayList<>();
+        lines.add("#!/bin/bash");
+        // NOTE: user data output is logged to `/var/log/cloud-init-output.log` automatically with ec2 instances
+        // Add some items to the $PATH as the $PATH with user-data scripts differs from the ssh $PATH.
+        lines.add("export PATH=\"$PATH:/home/ubuntu/.yarn/bin\"");
+        lines.add("export PATH=\"$PATH:/home/ubuntu/.nvm/versions/node/v12.16.3/bin\"");
+        // Remove previous files that might have been created during an Image creation
+        lines.add(String.format("rm %s/%s || echo '' > /dev/null", webDir, OTP_RUNNER_STATUS_FILE));
+        lines.add(String.format("rm %s || echo '' > /dev/null", otpRunnerManifestFile));
+        lines.add(String.format("rm %s || echo '' > /dev/null", manifest.jarFile));
+        lines.add(String.format("rm %s || echo '' > /dev/null", manifest.otpRunnerLogFile));
+        lines.add("rm /var/log/otp-build.log || echo '' > /dev/null");
+        lines.add("rm /var/log/otp-server.log || echo '' > /dev/null");
+
+        // FIXME what if the JSON has a single quote?
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            mapper.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+            lines.add(
+                String.format(
+                    "echo '%s' > %s",
+                    mapper.writeValueAsString(manifest),
+                    otpRunnerManifestFile
+                )
+            );
+        } catch (JsonProcessingException e) {
+            status.fail("Failed to create manifest for otp-runner!", e);
+            return null;
+        }
+        // install otp-runner
+        lines.add("yarn global add https://github.com/ibi-group/otp-runner.git");
```

Comment:
Does this need a fix me for referencing a release version or are we OK with just using the default branch's latest commit?

Reply:
I'm ok with just using github for now.

Reply:
I was more so thinking that we should do one of the two: 
- add some kind of note about how this will pull the latest commit on master (or whatever the default branch is) OR
- set up releases and specify a release version.